lib/AST/ASTImporter.cpp

```diff
@@ -4664,6 +4664,11 @@ Decl *ASTNodeImporter::VisitFunctionTemplateDecl(FunctionTemplateDecl *D) {
   if (!TemplatedFD)
     return nullptr;
 
+  // Check recursive import.
+  Decl *AlreadyImported = Importer.GetAlreadyImportedOrNull(D);
```

Comment:
I am wondering what should we do with this piece of code right now.
When we are importing the templated FunctionDecl above we will call import on the described template. This will result in a second call to importing the function template where the above import call will return the already imported function decl. After that we finish importing the template. After we imported the function decl and return from the function above we try to finish importing the same function template decl the second time. While having this check is definitely solving the problem I am wondering if there is a nicer solution for this cases in general. While I do agree it is good to fix this problem for now, maybe it worth a fixme to note that this is something that might have a better solution some day. @szepet  what do you think?